src/webgpu/api/operation/texture_view/format_reinterpretation.spec.ts

```diff
@@ -0,0 +1,390 @@
+      format,
+      size: [kTextureSize, kTextureSize],
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING,
+      viewFormats: [viewFormat],
+    });
+
+    t.device.queue.writeTexture(
+      { texture },
+      textureData,
+      {
+        bytesPerRow: texelSize * kTextureSize,
+      },
+      [kTextureSize, kTextureSize]
+    );
+
+    // Reinterepret the texture as the view format.
+    // Make a texel view of the format that also reinterprets the data.
+    const reinterpretedView = texture.createView({ format: viewFormat });
+    const reinterpretedTexelView = TexelView.fromTexelsAsBytes(viewFormat, inputTexelView.bytes);
+
+    // Create a pipeline to write data out to rgba8unorm.
+    const pipeline = t.device.createComputePipeline({
+      compute: {
+        module: t.device.createShaderModule({
+          code: `
+          @group(0) @binding(0) var src: texture_2d<f32>;
+          @group(0) @binding(1) var dst: texture_storage_2d<rgba8unorm, write>;
+          @stage(compute) @workgroup_size(1, 1) fn main(
+            @builtin(global_invocation_id) global_id: vec3<u32>,
+          ) {
+            var coord = vec2<i32>(global_id.xy);
+            textureStore(dst, coord, textureLoad(src, coord, 0));
+          }`,
+        }),
+        entryPoint: 'main',
+      },
+    });
+
+    // Create an rgba8unorm output texture.
+    const outputTexture = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: [kTextureSize, kTextureSize],
+      usage: GPUTextureUsage.STORAGE_BINDING | GPUTextureUsage.COPY_SRC,
+    });
+
+    // Execute a compute pass to load data from the reinterpreted view and
+    // write out to the rgba8unorm texture.
+    const commandEncoder = t.device.createCommandEncoder();
+    const pass = commandEncoder.beginComputePass();
+    pass.setPipeline(pipeline);
+    pass.setBindGroup(
+      0,
+      t.device.createBindGroup({
+        layout: pipeline.getBindGroupLayout(0),
+        entries: [
+          {
+            binding: 0,
+            resource: reinterpretedView,
+          },
+          {
+            binding: 1,
+            resource: outputTexture.createView(),
+          },
+        ],
+      })
+    );
+    pass.dispatch(kTextureSize, kTextureSize);
+    pass.end();
+    t.device.queue.submit([commandEncoder.finish()]);
+
+    const result = await textureContentIsOKByT2B(
+      t,
+      { texture: outputTexture },
+      [kTextureSize, kTextureSize],
+      {
+        expTexelView: TexelView.fromTexelsAsColors('rgba8unorm', reinterpretedTexelView.color, {
+          clampToFormatRange: true,
+        }),
+      },
+      { maxDiffULPsForNormFormat: 1 }
+    );
+    t.expectOK(result);
+  });
+
+g.test('render_and_resolve_attachment')
+  .desc(
+    `Test that a color render attachment allocated as 'format' is correctly rendered to as 'viewFormat',
+and resolved to an attachment allocated as 'format' viewed as 'viewFormat'.`
```

Comment:
```suggestion
and resolved to an attachment allocated as 'format' viewed as 'viewFormat'.

Other combinations aren't possible because the render and resolve targets must both match
in view format and match in base format.`
```